### PR TITLE
Various Java cleanups

### PR DESCRIPTION
**Pull request by mimaison**
- Removed usage of deprecated APIs
- Minor refactorings and code simplifications

### CODE REVIEW

File: src/main/java/io/strimzi/test/container/KafkaVersionService.java [hunk 1]
```diff
@@ -197,7 +197,7 @@ public String toString() {
      */
     public KafkaVersion latestRelease() {
         // at least one release in the json schema is needed
-        if (this.logicalKafkaVersionEntities == null || this.logicalKafkaVersionEntities.size() < 1) {
+        if (this.logicalKafkaVersionEntities.isEmpty()) {
```
Comment (mimaison) [inline src/main/java/io/strimzi/test/container/KafkaVersionService.java:200]: The field is `final` and initialized when declared so it should never be `null`.

File: src/test/java/io/strimzi/test/container/StrimziKafkaContainerIT.java [hunk 2]
```diff
@@ -72,7 +72,7 @@ void testStartContainerWithEmptyConfiguration(final String imageName) {
             systemUnderTest.start();
 
             assertThat(systemUnderTest.getBootstrapServers(), is("PLAINTEXT://"
-                    + systemUnderTest.getContainerIpAddress() + ":" + systemUnderTest.getMappedPort(9092)));
+                    + systemUnderTest.getHost() + ":" + systemUnderTest.getMappedPort(9092)));
```
Comment (mimaison) [inline src/test/java/io/strimzi/test/container/StrimziKafkaContainerIT.java:75]: `getContainerIpAddress()` is deprecated, so switching to `getHost()` which is the preferred method now.

File: src/test/java/io/strimzi/test/container/StrimziKafkaContainerMockTest.java [hunk 3]
```diff
@@ -18,8 +18,8 @@
 import static org.hamcrest.CoreMatchers.containsString;
 import static org.hamcrest.CoreMatchers.is;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.junit.Assert.assertTrue;
```
Comment (mimaison) [inline src/test/java/io/strimzi/test/container/StrimziKafkaContainerMockTest.java:21]: This is the junit 4 dependency brought via testcontainers. We should instead use `org.junit.jupiter.api.Assertions` which is junit 5 (the declared dependency in pom.xml).

**Review by see-quick — APPROVED** [review]
Thanks, please update the description of this PR.

**Comment by mimaison** [comment]
Thanks, I updated the description.

**Pull request merged**